element-connector/src/test/java/org/eclipse/californium/elements/tcp/TcpCorrelationTest.java

```diff
@@ -0,0 +1,323 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial implementation. 
+ ******************************************************************************/
+package org.eclipse.californium.elements.tcp;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.text.IsEmptyString.isEmptyOrNullString;
+import static org.junit.Assert.assertArrayEquals;
+
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.californium.elements.Connector;
+import org.eclipse.californium.elements.CorrelationContext;
+import org.eclipse.californium.elements.RawData;
+import org.eclipse.californium.elements.TcpCorrelationContext;
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+public class TcpCorrelationTest {
+
+	@Rule
+	public final Timeout timeout = new Timeout(20, TimeUnit.SECONDS);
+
+	private final List<Connector> cleanup = new ArrayList<>();
+
+	@After
+	public void cleanup() {
+		for (Connector connector : cleanup) {
+			connector.stop();
+		}
+	}
+
+	/**
+	 * Test, if the correlation context is determined proper.
+	 * Send a request and check, if the response has the same correlation context on the client side.
+	 * Send a second request and check, if this has the same correlation context on the client side.
+	 * Also check, if the server response is sent with the same context as the request was received.
+	 */
```

Comment:
Even it seems to be more a set of tests, I think, it's covering a normal use case "request-response-request" and therefore I put it into one.